From ebb60898724eb0568c13dd8a1db591da820ea010 Mon Sep 17 00:00:00 2001
From: Thomas Roos <throos@amazon.de>
Date: Wed, 21 Jan 2026 14:14:01 +0000
Subject: [PATCH] Fix ast.Str compatibility for Python 3.14+

Replace deprecated ast.Str with ast.Constant and a.value.s with a.value.value
for compatibility with Python 3.14+.

Upstream-Status: Inappropriate [embedded specific]
---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f84c128..41a6ab8 100644
--- a/setup.py
+++ b/setup.py
@@ -90,8 +90,8 @@ with open(join(here, "html5lib", "__init__.py"), "rb") as init_file:
         if (len(a.targets) == 1 and
                 isinstance(a.targets[0], ast.Name) and
                 a.targets[0].id == "__version__" and
-                isinstance(a.value, ast.Str)):
-            version = a.value.s
+                isinstance(a.value, ast.Constant)):
+            version = a.value.value
 
 setup(name='html5lib',
       version=version,
